### PR TITLE
Move out sendgrid email from airflow.contrib

### PR DESCRIPTION
**Pull request by mik-laj**
> Formerly the core code was maintained by the original creators - Airbnb. The code that was in the contrib
> package was supported by the community. The project was passed to the Apache community and currently the
> entire code is maintained by the community, so now the division has no justification, and it is only due
> to historical reasons.
> 
> To clean up, the `send_mail` function from the `airflow.contrib.utils.sendgrid` module has been moved.
> 
> If your configuration file looks like this.
> ```ini
> [email]
> email = airflow.contrib.utils.sendgrid.send_email
> ```
> It should look like this now:
> ```ini
> [core]
> executor = airflow.providers.sendgrid.utils.emai.send_email
> ```
> 
> The old configuration is still works but can be abandoned.
> 

In addition, I've moved documentation to /docs directory to make more people aware of this feature.

---
Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Target Github ISSUE in description if exists
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/utils/helpers.py [hunk 1]
```diff
@@ -195,18 +195,18 @@ def partition(pred: Callable, iterable: Iterable):
 
 
 def chain(*args, **kwargs):
-    """This module is deprecated. Please use `airflow.models.baseoperator.chain`."""
+    """This function is deprecated. Please use `airflow.models.baseoperator.chain`."""
```
Comment (potiuk) [inline airflow/utils/helpers.py]: Unrelated ?

File: docs/conf.py [hunk 2]
```diff
@@ -189,8 +189,6 @@
 exclude_patterns: List[str] = [
     # We only link to selected subpackages.
     '_api/airflow/index.rst',
-    # Required by airflow/contrib/plugins
```
Comment (potiuk) [inline docs/conf.py]: Also unrelated?

File: docs/integration.rst [hunk 3]
```diff
@@ -28,6 +28,7 @@ Airflow has a mechanism that allows you to expand its functionality and integrat
 * :doc:`Logging </howto/write-logs>`
 * :doc:`Tracking systems </howto/tracking-user-activity>`
 * :doc:`Secrets backends </howto/use-alternative-secrets-backend>`
+* :doc:`Email backends </howto/email-config>`
```
Comment (potiuk) [inline docs/integration.rst:31]: :+1:

**Review by potiuk — APPROVED** [review]
LGTM - just few unrelated small changes that might be better to split-off

File: UPDATING.md [hunk 4]
```diff
@@ -62,6 +62,28 @@ https://developers.google.com/style/inclusive-documentation
...
+
+If your configuration file looks like this.
+```ini
+[email]
+email_backend = airflow.contrib.utils.sendgrid.send_email
+```
+It should look like this now:
+```ini
+[email]
+email_backend = airflow.providers.sendgrid.utils.emai.send_email
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
email_backend = airflow.providers.sendgrid.utils.emailer.send_email
```

**Review by feluelle — CHANGES_REQUESTED** [review]

File: UPDATING.md [hunk 5]
```diff
@@ -62,6 +62,28 @@ https://developers.google.com/style/inclusive-documentation
...
+```
+
+The old configuration is still works but can be abandoned.
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
The old configuration still works but can be abandoned.
```

File: UPDATING.md [hunk 6]
```diff
@@ -62,6 +62,28 @@ https://developers.google.com/style/inclusive-documentation
...
+### SendGrid emailer has been moved
+
+Formerly the core code was maintained by the original creators - Airbnb. The code that was in the contrib
+package was supported by the community. The project was passed to the Apache community and currently the
+entire code is maintained by the community, so now the division has no justification, and it is only due
+to historical reasons.
+
+To clean up, the `send_mail` function from the `airflow.contrib.utils.sendgrid` module has been moved.
+
+If your configuration file looks like this.
```
Comment (feluelle) [inline UPDATING.md]:
```suggestion
If your configuration file looks like this:
```

**Review by feluelle — APPROVED** [review]

**Pull request merged**